code/modules/mob/living/carbon/human/human_emote.dm

```diff
@@ -0,0 +1,681 @@
+/datum/emote/living/carbon/human
+	/// Custom messages that should be applied based on species
+	/// Should be an associative list of species name: message
+	var/species_custom_messages = list()
+	/// Custom messages applied to mimes of a particular species
+	var/species_custom_mime_messages = list()
+	mob_type_allowed_typecache = list(/mob/living/carbon/human)
+
+/datum/emote/living/carbon/human/select_message_type(mob/user, msg, intentional)
+	. = ..()
+
+	var/mob/living/carbon/human/human_user = user
+
+	if(!species_custom_messages || (human_user.mind?.miming && !species_custom_mime_messages))
+		return .
+
+	var/custom_message
+	if(user.mind?.miming)
+		custom_message = species_custom_mime_messages[human_user.dna.species?.name]
+	else
+		custom_message = species_custom_messages[human_user.dna.species?.name]
+
+	if(custom_message)
+		return custom_message
+
+/datum/emote/living/carbon/human/run_emote(mob/user, params, type_override, intentional)
+	var/mob/living/carbon/human/H = user
+	if((emote_type & EMOTE_MOUTH) && !H.mind?.miming)
+		if(H.getOxyLoss() > 10 || H.AmountLoseBreath() >= 8 SECONDS)		// no screaming if you don't have enough breath to scream
+			H.emote("gasp")
+			return TRUE
+	. = ..()
```

Comment:
Yeah, so here the parent call comes at the end because we want to check at runtime whether or not the user can use the emote, and prevent the emote conditionally. Note that we can't just put the call in `can_run_emote` since that's what determines whether or not it shows up in `*help`. In this case in particular, if we put the `. = ..()` before, we'd get the standard emote message and *then* the gasp, of course.

It tends to pop up after some behavior when:
- We need to perform some runtime checks before sending the message
- We need to update the message or some other emote parameter before sending the emote

In most other cases, it tends to go towards the start of the proc with a check to see if the emote was executed successfully or not.

Reply:
I would rather `can_use_emote` and `can_run_emote` be split into separate procs to avoid the random positioning of the parent call, **BUT**, that is not a hill I am willing to die on. This PR needs merged already.

Reply:
Fair, that's mostly behavior from /tg/. Definitely something I could look at afterwards.